src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -878,6 +881,31 @@ err_t sockinfo_tcp::ip_output_syn_ack(struct pbuf *p, void* v_p_conn, int is_rex
 {
 	sockinfo_tcp *p_si_tcp = (sockinfo_tcp *)pcb_container;
 	p_si_tcp->m_p_socket_stats->tcp_state = new_state;
+
```

Comment:
The agent is using TCP sockinfo_tcp. So TCP should not know anything about the agent, (its internal members, state, etc) and should not call it. (it's like a lower layer calling an upper layer).
The way to do it is by the Agent registering in sockinfo_tcp for getting state updates (need to add this functionality into sockinfo_tcp) and having the Agent do its handling internally (not inside sockinfo_tcp)